### PR TITLE
Guide on Performance Management with Meshery

### LINKED ISSUE #3972: [Docs] Create "Performance Management with Meshery" guide
#### Current State
Users lack a holistic guide to use the performance management capabilities in Meshery.

This could be managed through the [UI](https://docs.meshery.io/guides/interpreting-performance-test-results), [CLI](https://docs.meshery.io/reference/mesheryctl/perf) and [GitHub Action](https://github.com/layer5io/meshery-smp-action).

#### Desired State
Create a new guide here: https://docs.meshery.io/guides

Add sections on performance management through UI, CLI as well as the GitHub action.

Refer: https://docs.meshery.io/guides/smi-conformance

---

#### Contributor [Guides](https://docs.meshery.io/project/contributing) and Resources
- 🛠 [Meshery Build & Release Strategy](https://docs.meshery.io/project/build-and-release)
- 📚 [Instructions for contributing to documentation](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#documentation-contribution-flow)
   - Meshery documentation [site](https://docs.meshery.io/) and [source](https://github.com/meshery/meshery/tree/master/docs)
- 🎨 Wireframes and designs for Meshery UI in [Figma](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI)
- 🙋🏾🙋🏼 Questions: [Layer5 Discussion Forum](https://discuss.layer5.io) and [Layer5 Community Slack](http://slack.layer5.io)

### PR DESCRIPTION
**Pull request by Anand-Theertha**
**Description**

This PR fixes #3972 

The guide adds three different ways the user can manage their performance tests namely - UI, CLI and GitHub Actions, with  appropriate snippets wherever necessary. 

**Notes for Reviewers**

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by l5io** [comment]
🚀 Preview for commit ca8522ec38b47405830153d1aa1a7e7f6f655b25 at: https://61423a3ead74c723436c9fb4--meshery-docs-preview.netlify.app

File: docs/pages/guides/performance-management.md [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+
+## Setup Meshery and run a Performance Test on your Service Mesh
+
+Install and login to Meshery to start running SMI conformance tests. See [Installation]({{ site.baseurl }}/installation) documentation for detailed steps on how to install Meshery.
+
+_Meshery dashboard_
+
+<a href="{{ site.baseurl }}/assets/img/smi/dashboard.png"><img alt="Meshery Dashboard" src="{{ site.baseurl }}/assets/img/smi/dashboard.png" /></a>
+
+Next, we navigate to the main Performance Testing dashboard. See [Performance Management]({{ site.baseurl }}/functionality/performance) to learn more about performance profiles, load generators and Kubernetes Cluster and Service Mesh Metrics.
```
Comment (warunicorn19) [inline docs/pages/guides/performance-management.md:19]:
```suggestion
Next, we navigate to the main Performance Testing dashboard. See [Performance Management]({{ site.baseurl }}/functionality/performance) to learn more about performance profiles, load generators, Kubernetes Cluster, and Service Mesh Metrics.
```

File: docs/pages/guides/performance-management.md [hunk 2]
```diff
@@ -0,0 +1,172 @@
+---
+layout: default
+title: Performance Management with Meshery
+description: This guide is to help users get a better overview of running and managing performance tests in Meshery
+permalink: guides/performance-management
+type: Guides
+---
+
+This guide is useful to learn how to manage performance tests with Meshery through the [UI](#performance-management-through-meshery-ui), [CLI](#performance-management-through-meshery-cli-(mesheryctl)) and a [GitHub Action](#performance-management-through-meshery-using-github-action) for your test results. To learn more about Meshery and Service Mesh Performance Management commands, see [Service Mesh Performance Management]({{ site.baseurl }}/reference/mesheryctl/commands/mesheryctl-perf)
```
Comment (warunicorn19) [inline docs/pages/guides/performance-management.md:9]:
```suggestion
This guide is useful to learn how to manage performance tests with Meshery through the [UI](#performance-management-through-meshery-ui), [CLI](#performance-management-through-meshery-cli-(mesheryctl)) and a [GitHub Action](#performance-management-through-meshery-using-github-action). To learn more about Meshery and Service Mesh Performance Management commands, see [Service Mesh Performance Management]({{ site.baseurl }}/reference/mesheryctl/commands/mesheryctl-perf)
```

File: docs/pages/guides/performance-management.md [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+
+The dashboard allows us to run our Performance Tests , it also shows the different tests run over the lifetime of the profile sorted by time and day. 
+
+## Performance Management Through Meshery UI
+
+<a href="{{ site.baseurl }}/assets/img/performance-management/performance-run-test.PNG"><img alt="Performance Test Options" src="{{ site.baseurl }}/assets/img/performance-management/performance-run-test.PNG" /></a>
+
+Now that we have our setup, we can run a performance test with our required constraints. We can modify multiple testing scenarios by altering the Service Mesh used, adjusting the Concurrent Requests taking place or the number of queries run or select a particular Load Generator. For the sake of simplicity, we can also personalize the name of the test for future reference. 
+
+Upon personalization, select **Run Test** to execute the current test or **Save Profile** to save the properties of the test and execute it at a convinient time.
```
Comment (warunicorn19) [inline docs/pages/guides/performance-management.md:33]:
```suggestion
Upon personalization, select **Run Test** to execute the current test or **Save Profile** to save the properties of the test and execute it at a convenient time. 
```

**Comment by Anand-Theertha** [comment]
I thought I had my signoff turned on for every commit, do I have to add in my signoff now?

**Comment by navendu-pottekkat** [comment]
Yes @Anand-Theertha You can check this: https://github.com/meshery/meshery/pull/4192/checks?check_run_id=3613570294

**Review by warunicorn19 — CHANGES_REQUESTED** [review]

**Comment by l5io** [comment]
🚀 Preview for commit 4cf340399075e5d8f5b6b5faec2374f39878d8ac at: https://61423ec0fa933f23cdcfcfdc--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 4cf340399075e5d8f5b6b5faec2374f39878d8ac at: https://61423ec44b991a247ef9a0fb--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit b64514a0f8bb98e83343b5e3d8dd0e5db0535e30 at: https://6142ff602187dbcc133caf87--meshery-docs-preview.netlify.app

File: docs/pages/guides/performance-management.md (hunk 2 above)
Comment (warunicorn19) [reply docs/pages/guides/performance-management.md:9]:
```suggestion
This guide is useful to learn how to manage performance tests with Meshery through the [UI](#performance-management-through-meshery-ui), [CLI](#performance-management-through-meshery-cli-(mesheryctl)) and a [GitHub Action](#performance-management-through-meshery-using-github-action) for your test results. To learn more about Meshery and Service Mesh Performance Management commands, see [Service Mesh Performance Management]({{ site.baseurl }}/reference/mesheryctl/perf)
```

File: docs/pages/guides/performance-management.md (hunk 1 above)
Comment (warunicorn19) [reply docs/pages/guides/performance-management.md:19]:
```suggestion
Next, we navigate to the main Performance Testing dashboard. See [Performance Management]({{ site.baseurl }}/functionality/performance-management) to learn more about performance profiles, load generators and Kubernetes Cluster and Service Mesh Metrics.
```

**Review by warunicorn19 — CHANGES_REQUESTED** [review]
Screenshot 2021-09-16 at 2 11 39 PM

The commands are not visible, this is probably because you've called it incorrectly, if you could please fix that. Everything else seems to be fine, good stuff👍.

File: docs/pages/guides/performance-management.md (hunk 2 above)
Comment (warunicorn19) [reply docs/pages/guides/performance-management.md:9]: broken link // @Anand-Theertha

File: docs/pages/guides/performance-management.md (hunk 1 above)
Comment (warunicorn19) [reply docs/pages/guides/performance-management.md:19]: Some broken links to look at. // @Anand-Theertha

**Comment by alphaX86** [comment]
> The commands are not visible, this is probably because you've called it incorrectly, if you could please fix that. Everything else seems to be fine, good stuff👍.

+1 to this, maybe try double checking the syntax you've typed on

File: docs/pages/guides/performance-management.md [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+_Viewing the Results_
+
+<a href="{{ site.baseurl }}/assets/img/performance-management/chart.png"><img alt="Performance Test Results" src="{{ site.baseurl }}/assets/img/performance-management/chart.png" /></a>
+
+## Performance Management Through Meshery CLI (mesheryctl)
+
+_Description_
+
+<!-- Description of the command. Preferably a paragraph -->
+{% assign name = site.data.mesheryctlcommands.cmds[page.command] %}
```
Comment (alphaX86) [inline docs/pages/guides/performance-management.md:62]: Oh wait... I found why it didn't show up you need to include a command field on top of this file.... I'll show you...

**Comment by Anand-Theertha** [comment]
> Screenshot 2021-09-16 at 2 11 39 PM
> 
> The commands are not visible, this is probably because you've called it incorrectly, if you could please fix that. Everything else seems to be fine, good stuff👍.

I was just seeking @navendu-pottekkat to solve this.

File: docs/pages/guides/performance-management.md [hunk 5]
```diff
@@ -0,0 +1,172 @@
+---
+layout: default
+title: Performance Management with Meshery
+description: This guide is to help users get a better overview of running and managing performance tests in Meshery
+permalink: guides/performance-management
+type: Guides
```
Comment (alphaX86) [inline docs/pages/guides/performance-management.md:6]:
```suggestion
layout: default
title: Performance Management with Meshery
description: This guide is to help users get a better overview of running and managing performance tests in Meshery
permalink: guides/performance-management
type: Guides
command: perf
```

**Comment by alphaX86** [comment]
Fix the DCO issue :)

File: docs/pages/guides/performance-management.md (hunk 4 above)
Comment (Anand-Theertha) [reply docs/pages/guides/performance-management.md:63]: Had the same doubt! Thanks!

**Comment by leecalcote** [comment]
Good progress, @Anand-Theertha 👍

**Comment by Anand-Theertha** [comment]
This PR has been closed and a new one has been opened for the same issue. [#4219]

**Pull request closed without merging**